tools/bumper/bumper.go

```diff
@@ -79,44 +79,54 @@ func main() {
 
 		if componentBumpNeeded {
 			logger.Printf("Bumping %s from %s to %s", componentName, currentReleaseTag, updatedReleaseTag)
-			// reset --hard git repo
-			exitWithError(fmt.Errorf("reset --hader repo not implemented yet"))
 
-			// Create PR
-			exitWithError(fmt.Errorf("create PR not implemented yet"))
-
-			// update components yaml in the bumping repo instance
-			componentsConfig, err := cnaoRepo.getComponentsConfig(inputArgs.componentsConfigPath)
-			if err != nil {
-				exitWithError(errors.Wrap(err, "Failed to get components config during bump"))
-			}
-
-			// update component's entry in config yaml
-			component.Commit = updatedReleaseCommit
-			component.Metadata = updatedReleaseTag
-			componentsConfig.Components[componentName] = component
-			err = cnaoRepo.updateComponentsConfig(inputArgs.componentsConfigPath, componentsConfig)
-			if err != nil {
-				exitWithError(errors.Wrap(err, "Failed to update components yaml"))
-			}
-
-			err = cnaoRepo.bumpComponent(componentName)
+			err = handleBump(cnaoRepo, component, componentName, inputArgs.componentsConfigPath, updatedReleaseTag, updatedReleaseCommit, proposedPrTitle)
 			if err != nil {
 				exitWithError(errors.Wrapf(err, "Failed to bump component %s", componentName))
 			}
-
-			// create a new branch name
-			branchName := strings.Replace(strings.ToLower(proposedPrTitle), " ", "_", -1)
-			logger.Printf("Opening new Branch %s", branchName)
-
-			// push branch to PR
-			exitWithError(fmt.Errorf("push branch to PR not implemented yet"))
 		} else {
 			logger.Printf("Bump not needed in component %s", componentName)
 		}
 	}
 }
 
+func handleBump(cnaoRepo *gitCnaoRepo, component component, componentName, componentsConfigPath, updatedReleaseTag, updatedReleaseCommit, proposedPrTitle string) (err error) {
+	defer func() {
+		err = cnaoRepo.resetRepo()
+	}()
```

Comment:
either do `defer cnaoRepo.resetRepo()` or 

```golang
defer cun() {
    err = cnaoRepo.resetRepo()
    exitWithError(errors.Wrap(err, ....
}
```

Reply:
not sure why?
this way if `cnaoRepo.resetRepo()` it'll return an err so it'll be caught in `handleBump`'s err catch:
```
			err = handleBump(cnaoRepo, component, componentName, inputArgs.componentsConfigPath, updatedReleaseTag, updatedReleaseCommit, proposedPrTitle)
			if err != nil {
				exitWithError(errors.Wrapf(err, "Failed to bump component %s", componentName))
			}
```

Reply:
I mean with the if statement

Reply:
Done